mongodb/terraform-provider-mongodbatlas
# feat: support new tags attribute in cluster, advanced_cluster, and serverless_instance

## Description

Ticket: [INTMDB-515](https://jira.mongodb.org/browse/INTMDB-515)

- Support new `tags` attribute in **cluster**, **advanced_cluster**, and **serverless_instance** resource and data sources.
- Includes deprecation for `labels` attribute defined in **cluster**, **advanced_cluster**.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)

## Further comments

## Testing comments

### Acceptance tests
- under the resource test file of cluster, advanced_cluster, and serverless_instance a new test case was added to verify creation and update of tags. These test cases include verification of the resource and data sources in the same test for time reasons (both running locally and in CI).

### Manual verification

- local binary was built and used to create a cluster, advanced_cluster, and serverless instance. Additional the respective data sources were used to get the created clusters and verify tags are present.
